vaadin-testbench-unit-shared/src/main/kotlin/com/vaadin/testbench/unit/internal/Shortcuts.kt

```diff
@@ -86,8 +115,13 @@ public fun Component._fireShortcut(key: Key, vararg modifiers: Key) {
     // contains a boolean value with key 'filter'. We need to fake the json object
     // as if it contained all filters (otherwise `matchesFilter()` would NPE on missing key)
     // and respond true only to the matching filter.
+
+    // we need to use hashableKeys not simple keys, as this is the implementation used in ShortcutRegistration
+    // class.
+    val hashableKeys = modifiers.map { getHashableKey(it) }
+
     @Suppress("UNCHECKED_CAST")
-    val data = MockFilterJsonObject(key, modifiers.toSet() as Set<KeyModifier>)
+    val data = MockFilterJsonObject(key, hashableKeys.toSet() as Set<KeyModifier>)
```

Comment:
Yes, Marco, I shared on Slack that the mentioned line changed, so my Monday/yd changes caused the difference between the testbench and flow, how they call this private method, which has triggered the test failures. I can definitely rewrite the flow part to help address the issue.

However, I would emphasize that generateEventModifierFilter() is always called with HashableKeys right now:
(and some other code parts making sure it)
```
    private void addKey(Key key) {
        assert key != null;

        HashableKey hashableKey = new HashableKey(key);

        if (Key.isModifier(key)) {
            if (!modifiers.contains(hashableKey)) {
                modifiers.add(hashableKey);
                prepareForClientResponse();
            }
        } else {
            if (primaryKey == null || !primaryKey.equals(hashableKey)) {
                primaryKey = hashableKey;
                prepareForClientResponse();
            }
        }
    }
```
Since generateEventModifierFilter() is a private method that is called through other methods and in a state when HashableKeys have already been added to the ShortcutRegistration object, it would make sense to call it properly with HashableKeys, just like in any other use case, it is not an open API it shall not be called with Key objects, and potentially method signature could have been changed as well to be less flexible and refer to this.

So another solution would be to make the flow side (the private method) more rigid (so use HashableKeys in the method contract, and parameters), to catch errors even before runtime, as this is a specific, exact use case for this private method.

I feel like we want to force the Liskov principle to private methods :D. 

**But as the consensus here as I feel is to make the methods more flexible, with less constraint, I can do that, let me make the PR for it.**

Worth mentioning that by doing this we can have more unexpected behavior in runtime as we are more flexible. 
But I am doing this just to share my opinion and ideas about the optimal solution.

I need a bit of time again to build the flow artifact, include it in a customer project, and manually as well trying out many combinations to make sure it is working properly (I always do manual tests too).

Reply:
So my TLDR again:D 

I think in general it makes sense for private methods to have them more rigid, and less flexible/accepting for input params to ensure correct behavior and catch incorrect working in compile time. My two cents! 🙇

Reply:
I thought I replied to this, sorry. 

Sure we can do this, I already did a commit containing the changes to do the HashAbleKey transformation within the MockFilterJsonObject! 🙇

Reply:
Done ✅. 
Please approve if you feel, have time and do not have any further request, comment. :)